### ISSUE TITLE
UnicodeDecodeError: 'charmap' codec can't decode byte 0x98 in position 6074: character maps to <undefined>

### ISSUE THREAD

**Issue by RussianButman** [csparpa/pyowm#210]
Help me please. What am I doing wrong?

```
from pyowm import OWM
owm = OWM('xxxxxxxx')
reg = owm.city_id_registry()
london_loc = reg.ids_for('London', 'GB')
print(london_loc)
```

**Comment by csparpa** [comment]
Hello @RussianButman can you please post here
  1 . the full stacktrace that you get
  2. what python version and PyOWM version are you using? It's enough to paste the outputs of the commands: `python -V` and `pip freeze | grep pyowm`

Thanks, this will help

This issue is looking very similar to https://github.com/csparpa/pyowm/issues/208 (if not exactly the same!)... anyway, I've tried to run your code in virtualenvs both on Python2.7 and on Python3.x but I got no errors :-S

**Comment by csparpa** [comment]
@RussianButman ping...

**Comment by mmagnuski** [comment]
I am having a similar error:
```
'charmap' codec can't decode byte 0x81 in position 1622: character maps to <undefined>
```

What I am doing:
```python
from pyowm import OWM
owm = OWM(my_API_key)
reg = owm.city_id_registry()
reg.ids_for('London')
```

Stack trace:
```
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
<ipython-input-20-8494084edaf6> in <module>()
      1 # looking for places with given name:
      2 reg = owm.city_id_registry()
----> 3 reg.ids_for('Warsaw')

C:\Anaconda3\lib\site-packages\pyowm\webapi25\cityidregistry.py in ids_for(self, city_name, country, matching)
     74         if country is not None and len(country) != 2:
     75             raise ValueError("Country must be a 2-char string")
---> 76         splits = self._filter_matching_lines(city_name, country, matching)
     77         return [(int(item[1]), item[0], item[4]) for item in splits]
     78 

C:\Anaconda3\lib\site-packages\pyowm\webapi25\cityidregistry.py in _filter_matching_lines(self, city_name, country, matching)
    149         else:
    150             filename = self._assess_subfile_from(city_name)
--> 151             lines = [l.strip() for l in self._get_lines(filename)]
    152 
    153         # look for toponyms matching the specified city_name and according to

C:\Anaconda3\lib\site-packages\pyowm\webapi25\cityidregistry.py in _get_lines(self, filename)
    197         res_name = resource_filename(__name__, filename)
    198         with gzip.open(res_name, "rt") as fh:
--> 199             lines = fh.readlines()
    200             if type(lines[0]) is bytes:
    201                 lines = map(lambda l: l.decode("utf-8"), lines)

C:\Anaconda3\lib\encodings\cp1250.py in decode(self, input, final)
     21 class IncrementalDecoder(codecs.IncrementalDecoder):
     22     def decode(self, input, final=False):
---> 23         return codecs.charmap_decode(input,self.errors,decoding_table)[0]
     24 
     25 class StreamWriter(Codec,codecs.StreamWriter):

UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 1622: character maps to <undefined>
```


I am using:

. | version
---|--- 
platform | Windows 10
python | Python 3.5.1 :: Anaconda 4.1.0 (64-bit)
pyowm | 2.7.1

**Comment by csparpa** [comment]
Hi @mmagnuski thanks for helping with this!

Please - if you can - take a look at the code in [this branch](https://github.com/csparpa/pyowm/tree/210_unicode_error) containing a possible fix for the issue...

In order to test it, you should first remove your current PyOWM installation with `pip3.5 uninstall pyowm` and then install again PyOWM by:

  1. Cloning the repo with `git clone https://github.com/csparpa/pyowm.git`
  2. Switching to the above-mentioned branch with `git checkout 210_unicode_error`
  3. Launching `python3.5 setup.py install`

Then you would be able to test the code snippet you posted and check if you get the issue again

I will test it also, but I need to get a Windows host

Thanks!

**Comment by csparpa** [comment]
Hi @mmagnuski and @RussianButman I've been able to test the branch on Windows 10 with Python3.5 and it seems to work fine (I manage to get the correct output for the `ids_for()` call when running mmagnusky's code snippet)

Can you please confirm that you also don't get exceptions?

**Comment by mmagnuski** [comment]
Thanks for detailed instructions, I installed with `develop` to make things easier in the future.
I can confirm the branch fixes the problem for me:
```python
reg.ids_for('London')
```
```
[(2643743, 'London', 'GB'), (4119617, 'London', 'US'), (4298960, 'London', 'US'), (4517009, 'London', 'US'), (5056033, 'London', 'US'), (5367815, 'London', 'US'), (6058560, 'London', 'CA')]
```
Thanks!

**Comment by csparpa** [comment]
Thanks @mmagnuski !

Fixed with #213
